enterprise/frontend/src/embedding-sdk/lib/theme/get-embedding-theme.ts

```diff
@@ -18,25 +18,23 @@ import {
 import type { MappableSdkColor } from "./embedding-color-palette";
 import { SDK_TO_MAIN_APP_COLORS_MAPPING } from "./embedding-color-palette";
 
-const getFontFamily = (theme: MetabaseTheme) =>
-  theme.fontFamily ?? DEFAULT_FONT;
-
-const SDK_BASE_FONT_SIZE = `${DEFAULT_SDK_FONT_SIZE / 16}em`;
+const SDK_BASE_FONT_SIZE = `${DEFAULT_SDK_FONT_SIZE}px`;
 
 /**
  * Transforms a public-facing Metabase theme configuration
  * into a Mantine theme override for internal use.
  */
 export function getEmbeddingThemeOverride(
   theme: MetabaseTheme,
+  font?: string,
 ): MantineThemeOverride {
   const components: MetabaseComponentTheme = merge(
     DEFAULT_EMBEDDED_COMPONENT_THEME,
     theme.components,
   );
 
   const override: MantineThemeOverride = {
-    fontFamily: getFontFamily(theme),
+    fontFamily: theme.fontFamily ?? font ?? DEFAULT_FONT,
```

Comment:
This is quite confusing why there are so many fonts. From my understanding:
- `theme` is what users passed to our SDK that might or might not include `font` so we need
- `font` this is from Metabase instance `font`, maybe it's better to name this `metabasteInstanceFont` or something
- `DEFAULT_FONT` then might not be needed since instance font should at least always be the fallback?

Reply:
You're correct it's confusing and we have too many fonts 😆 , I think the fist bit (`theme.fontFamily`) is not needed actually, we store the font from the theme in redux: https://github.com/metabase/metabase/pull/47764/files#diff-022f68f2a8bede328179d2d89b7d3e3d41fd78a95fcfa46e3a4cf78ca5535a36R58-R62

For a moment I had set the name to `fontFromInstance` or something like that, but I realized that was a lie too as the font from the getFontSelector is either the one from the embed slice (which is the one from the theme) or the one from the instance.
I also think the only reason we need `DEFAULT_FONT` is to use while loading the settings, as the application-fonts has a default value of Lato.

Let me try and see if we can remove `theme.fontFamily` and if so I'll add a couple of comments that hopefully will make it easier to understand what's going on

Reply:
>getFontSelector is either the one from the embed slice (which is the one from the theme) or the one from the instance.

The embed slice isn't set in the SDK though, since it's only called when we run Metabase frontend, in that sense `getFontSelector` on SDK is always an instance font (which might create even more confusion since it doesn't behave the same as in our Metabase FE).

Maybe creating a new selector entirely called `getInstanceFontSelector` might make things clearer. And we could use that in our SDK.

https://github.com/metabase/metabase/blob/54825a17312ee32702ea596fd20e3bdce3fee878/frontend/src/metabase/lib/embed.js#L45

Reply:
It shouldn't be used as the `embed` slice is supposed to be for interactive, but we're using it for the sdk as well at the moment :(
<img width="385" alt="Screenshot 2024-09-13 at 11 14 58" src="https://github.com/user-attachments/assets/f92a1508-0528-410d-90a2-bc428af29968">

We set the "embed options" from the sdk here: https://github.com/metabase/metabase/blob/16b470da02b08410b59b690b9505b335e35f2a41/enterprise/frontend/src/embedding-sdk/components/public/MetabaseProvider.tsx#L58-L62
We should change that but I'd keep it as a next step otherwise we'll never stop fixing stuff

Reply:
ok, this is another multi-layer code I really don't like since it's super hard to reason about :/ Thanks for pointing this out.

Reply:
Yeah, I really want to get this merged, write tests that don't depend on any implementation, and then we can think about refactoring and removing layers without being scared of breaking stuff

Reply:
@npretto `theme.fontFamily` is just for Mantine's components such as `Text` and `Title`, so it should always be identical to the Redux store, and therefore does not need to be referenced here. Just referencing the font from Redux as you did with `font` should be 100% okay.

Reply:
Does the code I just pushed look good?
I tested if mantine components use the instance font by default and they do, I'm honestly not sure which piece of code makes that work though 🙃

Reply:
@npretto that looks good! 🎉

Reply:
I tried to add `,sans-serif` to the default and I wanted to test it, but if i block the properties request, the page only renders the error components that for some reason don't have the correct fonts.
I feel like this is something that will take time to investigate and we'll have to add it to the "todo" for now